dev-java/cglib/cglib-3.1-r1.ebuild

```diff
@@ -0,0 +1,66 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+JAVA_PKG_IUSE="doc examples source test"
+
+inherit java-pkg-2 java-ant-2
+
+DESCRIPTION="cglib is a powerful, high performance and quality Code Generation Library"
+HOMEPAGE="https://github.com/cglib/cglib"
+SRC_URI="mirror://sourceforge/${PN}/${PN}-src-${PV}.jar"
+
+LICENSE="Apache-2.0"
+SLOT="3"
+KEYWORDS="~amd64 ~arm64 ~ppc64 ~x86"
+
+RESTRICT="!test? ( test )"
```

Comment:
i think this should be `RESTRICT="test"` if you want to disable the tests completely. with your syntax, it's still possible to run (and fail) tests for this package. i searched the devmanual but i must admit it does not provide enough info to use this feature in the right way, imo.

Reply:
Other proxy-maintainer advised me to use this in a recent case. Got many of examples searching:
`qgrep 'RESTRICT="!test?'`

Have you any idea how to fix this testing bug? [build.log.gz](https://github.com/gentoo/gentoo/files/6197077/build.log.gz)
Or should we really turn off the tests here?
Surprisingly with EAPI 5 and Java 11 it seems to work ...

Reply:
i was already looking at it before but i really have no idea what is it trying to say... anyway, it is a generally known problem, you can find some info for example [here](https://stackoverflow.com/questions/15122890/java-lang-verifyerror-expecting-a-stackmap-frame-at-branch-target-jdk-1-7#15122960). you can try to pass that `-noverify` to java for the tests to see if it has any effect.

Reply:
After trying many things now I prefer to use `RESTRICT="test"` as you suggested.  USE flag will go.

Reply:
>  think this should be `RESTRICT="test"` if you want to disable the tests completely. with your syntax, it's still possible to run (and fail) tests for this package. i searched the devmanual but i must admit it does not provide enough info to use this feature in the right way, imo.

@fordfrog 
Not in devmanual but found it mentioned [here](https://projects.gentoo.org/qa/policy-guide/other-metadata.html#pg0703). 
There seems to be another way to completely disable tests:  
From my experience, an ebuild not having the `test` will simply not run tests even with `FEATURES=test` set.
But imho all java ebuilds should have the test flag added to "doc source", except for packages coming without test cases.